table/block_based/block_based_table_reader.cc

```diff
@@ -595,12 +595,19 @@ Status BlockBasedTable::Open(
   Footer footer;
   std::unique_ptr<FilePrefetchBuffer> prefetch_buffer;
 
+  // Only retain read_options.deadline. In future, we may retain more
+  // options. Specifically, w ignore verify_checksums and default to
+  // checksum verification anyway when creating the index and filter
+  // readers.
+  ReadOptions ro;
+  ro.deadline = read_options.deadline;
+
   // prefetch both index and filters, down to all partitions
   const bool prefetch_all = prefetch_index_and_filter_in_cache || level == 0;
   const bool preload_all = !table_options.cache_index_and_filter_blocks;
 
   if (!ioptions.allow_mmap_reads) {
-    s = PrefetchTail(file.get(), file_size, force_direct_prefetch,
+    s = PrefetchTail(ro, file.get(), file_size, force_direct_prefetch,
```

Comment:
If the table open, and thus prefetching, happens in the context of a Get/MultiGet, it could cause the request to exceed the deadline. Also, I think we prefetch data that we're relatively certain will be used soon (for example index partitions), so it just shifts the latency around. 

Haven't considered performing prefetch in another thread. I think it already happens in background threads when the table open is initiated by VersionBuilder. If its for a foreground operation, not sure if we would gain much by doing it in another thread since the main operation would just block waiting for it to complete.